src/NCPopupMenu.cc

```diff
@@ -45,11 +53,17 @@ NCPopupMenu::NCPopupMenu( const wpos & at, YItemIterator begin, YItemIterator en
 	row[0] = item->label();
 	row[1] = item->hasChildren() ? "..." : "";
 
+	// if (item->isSeparator())
+	//     row[0] = "---";
+
 	YTableItem *tableItem = new YTableItem( row[0], row[1] );
 	// yuiDebug() << "Add to std::map: TableItem: " << tableItem << " Menu item: " << item << std::endl;
 
-	addItem( tableItem );
-	itemsMap[tableItem] = item;
+	NCTableLine::STATE state = item->isEnabled() ? NCTableLine::S_NORMAL : NCTableLine::S_DISABELED;
+
+	addItem( tableItem, state );
+
+	_items.push_back(new Item({tableItem, item}));
```

Comment:
What syntax is this? Please use the conventions of the existing code. Those curly braces obscur the intent of the code.